protocol-units/bridge/contracts/src/NativeBridge.sol

```diff
@@ -100,25 +134,44 @@ contract NativeBridge is AccessControlUpgradeable, PausableUpgradeable, INativeB
         uint256 amount,
         uint256 nonce
     ) internal {
-        // _l2l1RateLimit(amount);
+        _rateLimitInbound(amount);
+        // Ensure the bridge transfer has not already been completed
+        require(nonce > 0, InvalidNonce());
+        require(idsToInboundNonces[bridgeTransferId] == 0, CompletedBridgeTransferId());
         // Ensure the bridge transfer ID is valid against the initiator, recipient, amount, and nonce
         require(
             bridgeTransferId == keccak256(abi.encodePacked(initiator, recipient, amount, nonce)),
             InvalidBridgeTransferId()
         );
-        // Ensure the bridge transfer has not already been completed
-        require(noncesToIncomingBridgeTransferIds[nonce] == bytes32(0x0), CompletedBridgeTransferId());
 
         // Store the nonce to bridge transfer ID
-        noncesToIncomingBridgeTransferIds[nonce] = bridgeTransferId;
+        idsToInboundNonces[bridgeTransferId] = nonce;
 
         // Transfer the MOVE tokens to the recipient
         if (!moveToken.transfer(recipient, amount)) revert MOVETransferFailed();
 
         emit BridgeTransferCompleted(bridgeTransferId, initiator, recipient, amount, nonce);
     }
 
+    function setInsuranceFund(address _insuranceFund) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        insuranceFund = _insuranceFund;
+        emit InsuranceFundUpdated(_insuranceFund);
+    }
+
     function togglePause() external onlyRole(DEFAULT_ADMIN_ROLE) {
         paused() ? _pause() : _unpause();
+        emit PauseToggled(paused());
+    }
+
+    function _rateLimitOutbound(uint256 amount) internal {
```

Comment:
Please refer to #917 for a parameterization strategy. This allows for stopping the bridge without defunding the insurance fund. 

`4` is indeed the minimum denominator. THIS SHOULD BE COMMENTED.

Reply:
Contract is pausable, but will parametrize the denominator, I think the period is risky.

Reply:
The implementation provided in #917 is collision free. But, for safety, you would want to roll over balances, so leaving as `1 day` fixed is fine for now.